pkg/platform/runtime/setup/buildlog/buildlog.go

```diff
@@ -48,6 +48,16 @@ type Events interface {
 	Heartbeat(time.Time)
 }
 
+// BuildError designates a build log build error.
+type BuildError struct {
+	*locale.LocalizedError
+}
+
+// ProgressReportError designates an error in the event handler for reporting progress.
+type ProgressReportError struct {
```

Comment:
Can we name this "EventHandlerError"? The reason being the buildlog package has no concept or care of "progress reporting", all it cares about is invoking a handler.